helix-core/src/main/java/org/apache/helix/monitoring/metrics/WagedRebalancerMetricCollector.java

```diff
@@ -38,7 +41,11 @@
 
     // The following latency metrics are related to AssignmentMetadataStore
     StateReadLatencyGauge,
-    StateWriteLatencyGauge
+    StateWriteLatencyGauge,
+
+    // Count of any rebalance failure.
+    // Note the rebalancer may still be able to return an assignment based on the previous record on an error.
+    RebalanceFailureCount
```

Comment:
Do you think it would be more consistent to name it RebalanceFailureCount -> RebalanceFailureCounter as `Counter` is used in other classes like ZkClientPathMonitor?